### ISSUE TITLE
Which collections are supported?

### ISSUE THREAD

**Issue by littledan** [Agoric/proposal-readonly-collections#1]
In the shim, I just see Map and Set. However, I've heard lots of calls for read-only views on TypedArrays, and I imagine ordinary Arrays could also be an interesting case. Are these included in the proposal?

**Comment by erights** [comment]
@littledan yes. All enumerable collections, i.e., all collections other than `WeakMap` and `WeakSet`. `TypedArray` and `ArrayBuffer` are definitely included. I hope to include `Array` and `SharedArrayBuffer` as well, but these have additional challenges.

Although one can describe a normal object as a collection of property name to value mappings, for purposes of this proposal, I do not consider such to be collections. For purposes of this proposal, I also do not consider strings to be collections.

The shim only has `Map` and `Set` so far only because we can't do everything first ;)
